webpack/scenes/ContentViews/Details/ContentViewDetails.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { useSelector, shallowEqual } from 'react-redux';
+import { Grid, GridItem, TextContent, Text, TextVariants } from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+import PropTypes from 'prop-types';
+
+import DetailsContainer from './DetailsContainer';
+import ContentViewInfo from './ContentViewInfo';
+import { selectCVDetails } from './ContentViewDetailSelectors';
+import TabbedView from '../../../components/TabbedView/TabbedView';
+
+const ContentViewDetails = ({ match }) => {
+  const cvId = parseInt(match.params.id, 10);
+  const details = useSelector(state => selectCVDetails(state, cvId), shallowEqual);
+  const {
+    name,
+    label,
+    description,
+    composite,
+    force_puppet_environment: forcePuppetEnvironment,
+    solve_dependencies: solveDependencies,
+  } = details;
+  const info = {
+    name: {
+      value: name,
+      label: __('Name'),
+    },
+    label: {
+      value: label,
+      label: __('Label'),
+      editable: false,
+    },
+    description: {
+      value: description,
+      label: __('Description'),
+      textArea: true,
+    },
+    composite: {
+      value: composite,
+      label: __('Composite'),
+      boolean: true,
+      editable: false,
+    },
+    force_puppet_environment: {
+      value: forcePuppetEnvironment,
+      label: __('Force Puppet environment'),
+      boolean: true,
+      tooltip: __('With this option selected, a puppet environment will be created during publish ' +
+               'and promote even if no puppet modules have been added to the Content View'),
+    },
+    solve_dependencies: {
+      value: solveDependencies,
+      label: __('Solve dependencies'),
+      boolean: true,
+      tooltip: __('This option will solve RPM and Module Stream dependencies on every publish of ' +
+               'this Content View. Dependency solving significantly increases publish time ' +
+               '(publishes can take over three times as long) and filters will be ignored when ' +
+               'adding packages to solve dependencies. Also, certain scenarios involving errata ' +
+               'may still cause dependency errors.'),
+    },
+  };
+
+  const tabs = [
```

Comment:
Was about to request the same for this const, but realized the Patternfly API probably will make it impossible.  It's ok for this one, though, since it's more straightforward.

Reply:
ya, I'm a little torn on this, passing an array of objects and iterating on it is not the best pattern IMO (pf tables *cough* *cough*). I would rather be repetitive with small, declarative components.

I could make this into a `TabWrapper`  or `ForemanTab` that looks like `<TabWrapper title={title} ><MyComponent /></TabWrapper>` for each tab, so the TabWrapper could keep the styling and labels consistent

Reply:
sounds good to me!

Reply:
I actually couldn't get a wrapper component around a tab and still have it work well in the `Tabs` component. It was using anonymous functions, which didn't seem to be able to be recognized by `Tabs` as the active tab open. I tried different function syntax and hardcoding values but nothing worked. I think it has something to do with how it is implemented in patternfly.

I changed it back to the original tabs array that is iterated through. I think what would be best is to move this component to Foreman, detail out the issue and options, and open up the discussion to see if there are suggestions.